### PR TITLE
allow setting namespace for helm chart

### LINKED ISSUE #145: Override namespace installation
Hello everyone! I would like to propose to you to add a `namespaceOverride` parameter in `values.yaml` like it was done for cert-manager. 

In my case I have a single big chart, which used to install cert-manager, trust-manager and other stuff. In current configuration namespaced resources will be installed in release namespace, but I want to control that behavior.

### PR DESCRIPTION
**Pull request by vinny-sabatini**
This is based on cert-manager's helm chart override and resolves #145

### CODE REVIEW

**Comment by vinny-sabatini** [comment]
This should fix #198

**Comment by erikgb** [comment]
> This addresses #198

@vinny-sabatini, thanks for looking into this. I believe this PR should resolve https://github.com/cert-manager/trust-manager/issues/145? Can you please add a proper description to the PR?

File: deploy/charts/trust-manager/README.md [hunk 1]
```diff
@@ -56,6 +56,7 @@ Kubernetes: `>= 1.25.0-0`
 | image.repository | string | `"quay.io/jetstack/trust-manager"` | Target image repository. |
 | image.tag | string | `nil` | Target image version tag. Defaults to the chart's appVersion. |
 | imagePullSecrets | list | `[]` | For Private docker registries, authentication is needed. Registry secrets are applied to the service account |
+| namespace | string | `nil` | The namespace to deploy trust-manager resources into. |
```
Comment (erikgb) [inline deploy/charts/trust-manager/README.md:59]: I think you need to a an empty default (value), like it's done is cert-manager:

https://github.com/cert-manager/cert-manager/blob/9fd770a011815d8499a7fb35a6387982ec909762/deploy/charts/cert-manager/values.yaml#L98-L101

Also, this file is a generated file - generated with helm-docs. Which needs to be regenerated with `make update-helm-docs` after you have changed the values file.

**Review by erikgb — CHANGES_REQUESTED** [review]
Almost LGTM. Thanks for the description with the correct link.

File: deploy/charts/trust-manager/README.md (hunk 1 above)
Comment (vinny-sabatini) [reply deploy/charts/trust-manager/README.md]: Good catch, I believe I have covered those requested changes in 98cdbd3.

**Review by erikgb — APPROVED** [review]
/ok-to-test
/lgtm

**Review by erikgb — APPROVED** [review]

**Review by SgtCoDFish — APPROVED** [review]
/lgtm
/approve

Love it, thank you Vinny for fixing and Erik for reviewing 😁

**Pull request merged**